src/main/java/org/broadinstitute/hellbender/engine/Walker.java

```diff
@@ -0,0 +1,25 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+/**
+ * An abstract class to represent built-in walkers that inherit directly from {@link GATKTool}.
+ * Created by jonn on 6/28/18.
+ */
+public abstract class Walker extends GATKTool {
+
+    @Override
+    final protected ReferenceDataSource getReferenceDataSource() {
+        throw new GATKException("Should never access ReferenceDataSource in child classes of Walker.");
```

Comment:
Change to: "Walker tools outside of the engine package should not directly access the engine reference datasource. They should get their reference data via apply() instead. Tools that need direct datasource access (eg., to implement custom traversal patterns) should extend GATKTool directly rather than a walker class, or introduce a new walker base class for the new traversal."

Reply:
Fixed.